### PR TITLE
disttask: fix keep reporting error when cancel pending task

### LINKED ISSUE #48890: scheduler might keep reporting error when task cancelled before running
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
- start a import task
- dispatcher try move to next step, and save all subtasks into db(add subtask and update task is not in one txn, to handle the case that txn is large and we can add subtasks in batch)
- before dispatcher update task status and step, the task is cancelled
- dispatcher will generate revert subtasks for current step, which is stepInit
- import into didn't have subtask executor for init step, add index too.

```
[onError] [task-id=1] [error="unknown step -1 for import task 1"]
```

### 2. What did you expect to see? (Required)
cancelled
### 3. What did you see instead (Required)
scheduler keeps reporting error
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by D3Hunter**
### What problem does this PR solve?

Issue Number: close #48890

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by ywqzzy — APPROVED** [review]

File: pkg/disttask/framework/dispatcher/dispatcher.go [hunk 1]
```diff
@@ -480,16 +480,20 @@ func (d *BaseDispatcher) onErrHandlingStage(receiveErrs []error) error {
 }
 
 func (d *BaseDispatcher) dispatchSubTask4Revert(meta []byte) error {
-	instanceIDs, err := d.GetAllSchedulerIDs(d.ctx, d.Task)
-	if err != nil {
-		logutil.Logger(d.logCtx).Warn("get task's all instances failed", zap.Error(err))
-		return err
-	}
+	var subTasks []*proto.Subtask
+	// task might be cancelled when subtask dispatched but task state not updated.
```
Comment (okJiang) [inline pkg/disttask/framework/dispatcher/dispatcher.go:484]:
```suggestion
	// when step of task is `StepInit`, we do not need revert its subtasks.
```

How about this?

**Review by okJiang — APPROVED** [review]

File: pkg/disttask/framework/dispatcher/dispatcher.go [hunk 2]
```diff
@@ -480,16 +480,20 @@ func (d *BaseDispatcher) onErrHandlingStage(receiveErrs []error) error {
...
+	if d.Task.Step != proto.StepInit {
```
[the remark this answers is not in the document]
Comment (D3Hunter) [reply pkg/disttask/framework/dispatcher/dispatcher.go:485]: see updated comment

File: pkg/disttask/framework/dispatcher/dispatcher.go [hunk 3]
```diff
@@ -480,16 +480,20 @@ func (d *BaseDispatcher) onErrHandlingStage(receiveErrs []error) error {
 }
 
 func (d *BaseDispatcher) dispatchSubTask4Revert(meta []byte) error {
-	instanceIDs, err := d.GetAllSchedulerIDs(d.ctx, d.Task)
-	if err != nil {
-		logutil.Logger(d.logCtx).Warn("get task's all instances failed", zap.Error(err))
-		return err
-	}
+	var subTasks []*proto.Subtask
+	// when step of task is `StepInit`, we do not need revert its subtasks.
```
Comment (D3Hunter) [inline pkg/disttask/framework/dispatcher/dispatcher.go:484]:
```suggestion
	// when step of task is `StepInit`, no need to do revert
```

**Pull request merged**